HTTPServerProject.Tests/HTTPServerWriteTests/WriteRequestTests.cs

```diff
@@ -0,0 +1,85 @@
+using HTTPServerRead.Header;
+using HTTPServerRead.Body;
+using HTTPServerReadTests.Streams;
+using HTTPServerWriteTests.Streams;
+using HTTPServerWrite.Request;
+using HTTPServerWrite.Response;
+using HTTPServerProxy.Response;
+
+namespace HTTPServerWriteTests.RequestTests;
+
+public class UnitTestsWritingRequests
+{
+
+    public List<string> request = new List<string>() 
+	{ 
+		"GET / HTTP/1.1", 
+		"Host: localhost:5000", 
+		"User-Agent: curl/7.79.1", 
+		"Accept: */*", 
+		"", 
+		"quit" 
+	};
+
+    [Fact]
+    public void GetProxyRequestTest()
+    {
+        List<string> expected = new List<string>() 
+		{ 
+			"GET / HTTP/1.1", 
+			"Host: localhost:5000", 
+			"User-Agent: curl/7.79.1", 
+			"Accept: */*", 
+			"", 
+			"quit" 
+		};
+        var reader = new TestReadStreams(request);
+        var header = new Header(reader);
+        var initialLine = header.GetLine();
+        var reqHeaders = header.GetHeaders();
+        var body = new Body(reader);
+        var reqBody = body.GetBody();
+
+        var proxyStream = new List<string>();
+        var writer = new TestWriteStreams(proxyStream);
+
+        var proxyRequest = new WriteRequest(writer: writer, initialLine: initialLine, headers: reqHeaders, body: reqBody);
+        proxyRequest.GetRequest();
+
+        Assert.Equal(expected, proxyStream);
+    }
+
+    [Fact]
+    public void GetProxyResponseTest()
+    {
+        var expected = new List<string>() 
+		{ 
+			"HTTP/1.1 200 OK", 
+			"Allow: GET, OPTIONS", 
+			"", 
+			"Hello World" 
+		};
+        var stream = new List<string>();
+        var proxyStream = new List<string>() 
+		{ 
+			"HTTP/1.1 200 OK", 
+			"Allow: GET, OPTIONS", 
+			"", 
+			"Hello World" 
+		};
+        var proxyReader = new TestReadStreams(proxyStream);
+
+        var proxyResponseHeader = new Header(proxyReader);
+        var proxyStatusLine = proxyResponseHeader.GetLine();
+        var proxyStatusCode = proxyResponseHeader.GetCode(proxyStatusLine);
+        var pRHeaders = proxyResponseHeader.GetHeaders();
+        var proxyResponseBody = new Body(proxyReader);
+        var pRBody = proxyResponseBody.GetBody();
+
+        var writer = new TestWriteStreams(stream);
+        var response = new WriteResponse(writer: writer, code: proxyStatusCode, body: pRBody, headers: pRHeaders);
+        response.GetResponse();
+
+        Assert.Equal(expected, stream);
+    }
+}
```

Comment:
I don't understand - what do these tests have to do with the proxy?

Reply:
I updated the naming of the test. I also removed the `GetProxyResponseTest` because it has nothing to do with the `WriteRequest` class